plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/XmlSourceValidatorTest.java

```diff
@@ -107,21 +107,26 @@ public void testValidate_dynamicWebProject() throws CoreException, ValidationExc
   }
 
   @Test
-  public void testValidate_noProblemElements() throws IOException {
+  public void testValidate_noProblemElements() throws IOException, CoreException {
     XmlSourceValidator validator = new XmlSourceValidator();
     validator.setHelper(new AppEngineWebXmlValidator());
-    byte[] xml = ("<appengine-web-app xmlns='http://appengine.google.com/ns/1.0'>"
+    String xml = "<appengine-web-app xmlns='http://appengine.google.com/ns/1.0'>"
         + "<runtime>java8</runtime>"
-        + "</appengine-web-app>").getBytes(StandardCharsets.UTF_8);
-    validator.validate(reporter, null, xml);
+        + "</appengine-web-app>";
+    
+    IProject project = appEngineStandardProject.getProject();
+    IFile file = project.getFile("testdata.xml");
+    file.create(ValidationTestUtils.stringToInputStream(APPLICATION_XML), 0, null);
+    
+    validator.validate(reporter, file, xml.getBytes(StandardCharsets.UTF_8));
```

Comment:
`file` is `testdata.xml`, whose content is `APPLICATION_XML`. Also, the `xml.getBytes()` content is different from `APPLICATION_XML`. It is not clear if this validates the content of `testdata.xml` or `xml`. Maybe the code should be `file.create(stringToInputStream(xml), 0, null)`?

Reply:
done